Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/EntryDelegate.cpp

```diff
@@ -138,38 +139,61 @@ namespace AzToolsFramework
             m_showSourceControl = showSourceControl;
         }
 
-        int EntryDelegate::DrawThumbnail(QPainter* painter, const QPoint& point, const QSize& size, Thumbnailer::SharedThumbnailKey thumbnailKey) const
+        int EntryDelegate::DrawThumbnail(QPainter* painter, const QPoint& point, const QSize& size, const AssetBrowserEntry* entry) const
         {
-            SharedThumbnail thumbnail;
-            ThumbnailerRequestBus::BroadcastResult(thumbnail, &ThumbnailerRequests::GetThumbnail, thumbnailKey);
-            AZ_Assert(thumbnail, "The shared numbernail was not available from the ThumbnailerRequestBus.");
-            AZ_Assert(painter, "A null QPainter was passed in to DrawThumbnail.");
-            if (!painter || !thumbnail || thumbnail->GetState() == Thumbnail::State::Failed)
+            if (!m_showSourceControl)
             {
-                return 0;
-            }
-
-            const Thumbnail::State thumbnailState = thumbnail->GetState();
-            if (thumbnailState == Thumbnail::State::Loading)
-            {
-                AzQtComponents::StyledBusyLabel* busyLabel;
-                AssetBrowserComponentRequestBus::BroadcastResult(busyLabel , &AssetBrowserComponentRequests::GetStyledBusyLabel);
-                if (busyLabel)
+                const auto& qVariant = AssetBrowserViewUtils::GetThumbnail(entry);
+                if (const auto& path = qVariant.value<QString>(); !path.isEmpty())
                 {
-                    busyLabel->DrawTo(painter, QRectF(point.x(), point.y(), size.width(), size.height()));
+                    QIcon icon;
+                    icon.addFile(path, size, QIcon::Normal, QIcon::Off);
+                    AZ_Assert(!icon.isNull(), "Asset Browser Icon not found for file '%s'", path.constData());
```

Comment:
repeat: This needs to convert the QString to a utf8 null-terminated string.
```suggestion
                    AZ_Assert(!icon.isNull(), "Asset Browser Icon not found for file '%s'", path.toUtf8().constData());
```

Reply:
Thanks, fixed!